spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ehcache/EhCacheAutoConfiguration.java

```diff
@@ -0,0 +1,85 @@
+package org.springframework.boot.autoconfigure.ehcache;
+
+import net.sf.ehcache.CacheManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cache.ehcache.EhCacheCacheManager;
+import org.springframework.cache.ehcache.EhCacheManagerFactoryBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.util.Assert;
+
+import javax.annotation.PostConstruct;
+
+/**
+ * {@link org.springframework.boot.autoconfigure.EnableAutoConfiguration Auto-configuration} for EhCache.
+ *
+ * @author Eddú Meléndez
+ * @since 1.3.0
+ */
+@Configuration
+@ConditionalOnClass(CacheManager.class)
+public class EhCacheAutoConfiguration {
+
+    @Configuration
+    @ConditionalOnMissingBean(EhCacheManagerFactoryBean.class)
+    @EnableConfigurationProperties(EhCacheProperties.class)
+    public static class EhCacheManagerFactoryConfiguration {
+
+        @Autowired
+        private EhCacheProperties properties;
+
+        @Autowired
+        private ResourceLoader resourceLoader;
+
+        @PostConstruct
+        public void checkLocationExists() {
+            if (this.properties.isCheckLocation()) {
+                Assert.state(!this.properties.getLocation().isEmpty(),
+                    "EhCache config file location is not configured");
+                boolean exists = hasLocation();
+                Assert.state(exists, "Cannot find ehcache config file location in: "
+                    + this.properties.getLocation()
+                    + " (please add ehcache config file or check your EhCache configuration)");
+            }
+        }
+
+        private boolean hasLocation() {
+            String location = this.properties.getLocation();
+            if (this.resourceLoader.getResource(location).exists()) {
+                return true;
+            }
+            return false;
+        }
+
+        @Bean
+        public EhCacheManagerFactoryBean ehCacheManagerFactoryBean() {
+            EhCacheManagerFactoryBean ehCacheFactoryBean =
```

Comment:
I would use `EhCacheManagerUtils` and I'd expose the `CacheManager` directly and not the factory bean.